solutions/CSES Swap Game.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: cses-1670
+source: CSES
+title: Swap Game
+author: Maggie Liu
+---
+
+## Explanation
+
+Since there are only $9!$ states the grid can be in, we can use BFS to find the
+number of moves needed to reach the sorted grid from the starting grid. We
+represent each $\texttt{State}$ as the $\texttt{grid}$: an array of $9$ numbers,
+and the $\texttt{dist}$: the number of moves needed to reach this
+$\texttt{grid}$.
+
+For each $\texttt{State}$, we can swap two horizontally adjacent squares or two
+vertically adjacent squares. If we haven't processed this new grid yet, we can
+push it into the queue and mark it as visited in the $\texttt{vis}$ array. When
+we reach the sorted grid, we print the $\texttt{dist}$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2!)$, where $N$ is the side length of the
+grid
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <queue>
+#include <array>
+#include <algorithm>
+using namespace std;
+
+struct State
+{
+	array<int, 9> grid;
+	int dist;
+};
+
+bool vis[9][9][9][9][9][9][9][9][9];
```

Comment:
Could you try encoding each grid as a single base 9 number please? I think it'll look much neater that way